### ISSUE TITLE
"Add server" dialog should be empty by default, autofilling explicitly

### ISSUE THREAD

**Issue by chauffer** [mumble-voip/mumble#2303]
Right now it seems to keep the last added server's config; which isn't ideal since the default port is hard to remember (64738) yet it's configured on most servers.

**Comment by mkrautz** [comment]
This seems like a bug to me.

**Comment by chauffer** [comment]
Connect - Add New on latest snapshot (1256) seems to have my last added server's configuration.

Is it supposed to be empty?

**Comment by mkrautz** [comment]
No, that happens to me, too.

However, I don't think it should.

**Comment by bendem** [comment]
Heh, I always thought it was a (not so useful) feature.

**Comment by mkrautz** [comment]
I can't reproduce this anymore?

Does it still happen for you? @bendem @chauffer

**Comment by bendem** [comment]
I can still reproduce on current master

**Comment by mkrautz** [comment]
Actually, what happens here is seemingly that if you have a mumble:// URL in your pasteboard, it will auto-fill the dialog.

So, not the last server you added, but what's in your pasteboard.

Agree?

Or, if not, please provide steps to reproduce. :-)

**Comment by bendem** [comment]
Well, I don't have a mumble url in my clipboard. I just open the server list, click on "Add New" and the dialog that opens is already populated.

**Comment by bendem** [comment]
More info on this, the "add server" dialog is only fully populated if I'm connected to a server (with the information of that server).
If I'm not connected, only the port is populated with the port of one of my saved servers (always the same but I could not figure out the logic).

**Comment by bendem** [comment]
@mkrautz So is this a bug? From the code it looks like intended behavior.

**Comment by hacst** [comment]
It is intended behavior. The idea was that if you joined a server via a mumble:// url you are quite likely to want to add it to your favourites. Hence the pre-fill for convinience.

**Comment by mkrautz** [comment]
OK, so the connect dialog has the following behaviors:

 - If you're connected via a mumble:// URL, it will auto-fill that URL
 - If you have a mumble:// URL in your clipboard, it will auto-fill that URL

Hmmm....

**Comment by hacst** [comment]
Yeah. Pretty much optimized for "if you are handed/clicked a url and then want to add the server". Iirc I assumed that would be the usual cases for when you add a server (if the admin knows of mumble:// urls that is ;) ).

**Comment by mkrautz** [comment]
I do remember the functionality being added, but as a user, it's surprising.

Perhaps a QMessageBox asking you whether you want to auto-fill the currently connected server (if connected via mumble://), or auto-fill from the clipboard (if a mumble:// URL is in the clipboard)?

**Comment by bendem** [comment]
What about this?
+ if a mumble:// url is in the clipboard, autofill with that
+ if the server is already in the favorites, leave fields blank
+ if none of the above apply, fill the fields with the current server's data.

I find solving UX problems by adding dialogs is generally not the right solution.

**Comment by mkrautz** [comment]
The problem with the feature as-is is that it was indistinguishable from a bug to me.

I'm fine with auto-filling as long as the user is made aware that it has happened.
Perhaps small notification

**Comment by mkrautz** [comment]
@bendem I agree that adding dialogs is not the correct solution.
However, in QtWidgets, that's often the most straight forward solution.
We're often blocked on things due to it being hard to express the UI we want.

For example, the local user volume feature was blocked for a long time. It was initially a slider in a sub-menu of a user's context menu.

In the end, it was converted to a dialog.

I wanted the other approach, but it stalled.

Dialogs were easier.

**Comment by bendem** [comment]
True, the lack of manpower is always a problem. So easier is better from a development POV

**Comment by Tarun80** [comment]
My friend and I were discussing this yesterday. Based upon the ideas we tossed back and forth, we spoke about changing some things as follows:

* Change the Add New... button to just Add... (Or Add Server...)
* When clicking on the Add... button, all fields are empty. 
* If the user wants to add a new server, there's a checkbox for "Add current?" which can then check the clipboard or pull data from the server you're connected to. Perhaps there could also be a check where if text has been entered into any of the above text fields, it warns that it may be overwritten when they click "Add current?"
* Additionally, under the Server menu on the client, an "Add to Favorites" option.

We had also discussed ideas for a prompt after connecting. However that could also cause issues with too many prompts, especially after connecting to a server not using an SSL certificate. Also, if their first use it would mean Audio Wizard, Adding a server, no SSL prompt if it applies, then "Add to Favorites?" which would be overwhelming the user.

We both agreed that it would be a bad idea that could be abused by others to have something like...
```
mumble://exampleserver.com?title=MyAwesomeServer&version=1.2.0&addfavorite=yes
```

**Comment by hacst** [comment]
@mkrautz  Having a info-bar style message about the auto-fill (maybe with a reset button to undo what it did) sounds like a decent solution.

I really don't want additional dialogs, popups or additional potentially confusing ui elements in there if we can help it at all.

**Comment by Tarun80** [comment]
If I'm connected to a server and I need to add another server, I really don't want to have to hit reset every time I need to add a new server. Those fields should be blank by default, in my opinion.

**Comment by hacst** [comment]
@Tarun80 We could prevent pre-fill if the server is already a favorite.

**Comment by mkrautz** [comment]
Removing the 1.3.0 milestone since this is not a bug.

**Comment by Kissaki** [comment]
IMO the dialog fields should be empty, and filling it with the current server/clipboard data should be an explicit action.

Additional buttons would be an option, greyed out if not available.
UX wise better would be checking for available data and suggesting to the user to fill it in. This way the text could elaborate on the suggestion.

**Comment by Kissaki** [comment]
This should be resolved by #3182 once it lands.
It offers filling the dialog data with clipboard data if available, and if not with the current connected to server data if connected.

**Comment by davidebeatrici** [comment]
Resolved by #3182.
